src/ReactiveDomain/Tools/UpdateNugetVersion.ps1

```diff
@@ -1,34 +1,110 @@
+# UpdateNugetVersion.ps1: Build script to modify reactiveDomain .nuspec file
+#			  Packages.config file will be checked for the correct
+#			  Nuget version and will insert that version in the nuspec file	
+#
+# Author: Allen Thurman
+#
+# Revision: As Issued
+#
 
 $currentScriptDirectory = Get-Location
 [System.IO.Directory]::SetCurrentDirectory($currentScriptDirectory.Path)
 
 Set-Location -Path ${PSScriptRoot}
 $solutiondir = Resolve-Path -Path "$currentScriptDirectory\src\ReactiveDomain"
 
+$packages_Config = "$solutiondir\ReactiveDomain\packages.config"
+Write-Host "UpdateNuGetVersion:  Packages configuration file is: " $packages_Config
+
 Write-Host "Solution dir is " $solutiondir
 
 $path = "$solutiondir\ReactiveDomain\ReactiveDomain.nuspec"
 Write-Host "NuSpec File is" $path
 
 $dll = "$solutiondir\ReactiveDomain\bin\x64\Debug\ReactiveDomain.dll" 
-Write-Host "Loading dll: " $dll
+Write-Host "Loading ReactiveDomain dll: " $dll
 
 $Assembly = [Reflection.Assembly]::Loadfile($dll)
 $AssemblyName = $Assembly.GetName()
 $Assemblyversion = $AssemblyName.version.ToString()
 
-Write-Host "Assembly version is" $Assemblyversion
+Write-Host 
+Write-Host "ReactiveDomain Assembly version is" $Assemblyversion
+
+# Get version of the Dependent packages that are installed
+$packagexml = [xml](Get-Content $packages_Config)
+
+$ES = $packagexml.SelectSingleNode('//packages/package[@id="EventStore.Client"]')
```

Comment:
Not really because we have to explicitly define the dependencies somewhere: The closest thing would be to make it check the existing .nuspec file for the dependencies and then find their corresponding version in the packages.config. This would mean changing the base .nuspec file every time we add or remove a package instead. In any case, I don't see any way around some manual step if we add/remove a package